src/pages/resultsView/enrichments/EnrichmentsDataSetDropdown.tsx

```diff
@@ -51,41 +50,22 @@ export default class EnrichmentsDataSetDropdown extends React.Component<
         }
     }
 
-    @computed private get molecularProfileMap() {
-        if (this.props.dataSets.isComplete) {
-            return _.keyBy(
-                this.props.dataSets.result!,
-                x => x.molecularProfileId
-            );
-        }
-        return {};
-    }
-
     @computed private get studiesMap() {
         return _.keyBy(this.props.studies, x => x.studyId);
     }
 
-    @computed private get showEnrichmentsDataSetDropdown() {
```

Comment:
why did you remove these two computeds? seems like it will lead to unnecessary recomputation

Reply:
The reason I removed those two computes is: The `this.props.dataSets` is not a promise anymore, so when we deal with them, we don't need to see if the dataSets is loading completed. They may don't need to be recomputed. I moved the logic below from L58-L65

Reply:
I see, yes but the @computed is not just about checking the promise status, it's also about saving on recomputation as long as dataSets doesn't change. the way you have it now, it will recompute in every render which is unnecessary

Reply:
@adamabeshouse Thanks, that's right. I will bring the compute back.